### ISSUE TITLE
API tests fails

### ISSUE THREAD

**Issue by agentmilindu** [scorelab/Bassa#513]
## Expected Behavior
The tests should pass

## Current Behavior
![screenshot from 2018-10-07 00-05-35](https://user-images.githubusercontent.com/1391468/46574722-857b7100-c9c5-11e8-9aa2-a39ec7aa0a44.png)

## Steps to Reproduce (for bugs)
1. checkout to `develop` branch
2. cd to `components/core`
3. run `python3 setup.py test`

## Your Environment
* Version used: v1.0.0
* Environment name and version: Node v9.11.2
* Operating System and version: Ubuntu 18.04 LTS

**Comment by VibhorCodecianGupta** [comment]
@agentmilindu what would this signify?

![screen shot 2018-10-07 at 12 24 59 am](https://user-images.githubusercontent.com/29275810/46574831-80b7bc80-c9c7-11e8-8107-e6cad6b71eac.png)

**Comment by agentmilindu** [comment]
I think this is file permissions, try changing the `chmod`

**Comment by VibhorCodecianGupta** [comment]
Yes, I just did. I ran this with `sudo`, it worked. `chmod` would be a better option though.

**Comment by agentmilindu** [comment]
@VibhorCodecianGupta You could fix the issue?

**Comment by VibhorCodecianGupta** [comment]
Yes, will do!

**Comment by VibhorCodecianGupta** [comment]
Working on this, hopefully tonight :)

**Comment by VibhorCodecianGupta** [comment]
@agentmilindu @vivonk from what I gather, it was some kind of an import issue. The screenshot you posted got resolved, so I'm assuming that particular error got resolved (it's highly likely that its not though). This is what I did in `setup.py` (after renaming the test files by replacing hyphens with underscores, which were otherwise throwing a syntax error) -

![screen shot 2018-10-08 at 12 38 37 am](https://user-images.githubusercontent.com/29275810/46585784-e329bf00-ca92-11e8-8971-9664ebab9260.png)

However, after doing this and running the commands, this pops up -

![screen shot 2018-10-08 at 12 51 36 am](https://user-images.githubusercontent.com/29275810/46585870-5f70d200-ca94-11e8-8f1c-66fbb44785b2.png)

which basically means `requests` package has problem in importing/downloading. So I tried downloading the package using `easy_install` as well as both `pip` and `pip3`. Still, nothing worked, with the same error. I can't seem to figure out as to why even after downloading the package, it shows that requests is not found. I tried uninstalling `requests` using both `pip` and `pip3`, and here is the screenshot (I rejected the uninstall):

![screen shot 2018-10-08 at 12 59 36 am](https://user-images.githubusercontent.com/29275810/46585968-7d8b0200-ca95-11e8-9299-8a451275aec1.png)

This shows that `requests` is installed in both versions of python, i.e `2.7` and `3.7`. Still, the package is not loading when running the script. Could it be that it is not identifying which version to use, or maybe is using the wrong version? I'm new to python here, could you walk me through this?

**Comment by VibhorCodecianGupta** [comment]
@vivonk #516 has what you requested.

**Comment by VibhorCodecianGupta** [comment]
@agentmilindu any workarounds you have in mind? Or perhaps I'm doing something wrong? This seems to be an environment issue on my machine only, would love to get some solutions to this

**Comment by VibhorCodecianGupta** [comment]
@agentmilindu So I was able to resolve the issue. Turns out everything is working absolutely fine. Just the names of the test files needed to be changed. The `unittest` module recognises files of the type `test_*.py` OR `*_test.py`. Hence, changed the names accordingly. To run the tests, the python server should be running and database connection should be open, and the test files need to be imported from a python shell. Everything works fine then. Attaching screenshot.

![screen shot 2018-11-04 at 9 26 11 am](https://user-images.githubusercontent.com/29275810/47959868-862f1380-e014-11e8-8328-1aa632d6beb0.png)
